src/main/cpp/xmlhw/CanCoderDefn.cpp

```diff
@@ -0,0 +1,56 @@
+#include <memory>
+#include <string>
+#include <xmlhw/CanCoderDefn.h>
+#include <pugixml/pugixml.hpp>
+#include <utils/Logger.h>
+#include <ctre/Phoenix.h>
+#include <ctre/phoenix/sensors/CANCoder.h>
+
+using namespace std;
+using namespace frc;
+using namespace pugi;
+
+std::shared_ptr<ctre::phoenix::sensors::CANCoder> CanCoderDefn::ParseXML
+(
+    pugi::xml_node CanCoderNode
+)
+{
+    shared_ptr<ctre::phoenix::sensors::CANCoder> cancoder = nullptr;
+
+    string usage;
+    int canID = 0;
+    
+    bool hasError = false;
+
+    for(pugi::xml_attribute attr = CanCoderNode.first_attribute(); attr &&!hasError; attr = attr.next_attribute() )
+    {
+        if ( strcmp( attr.name(), "usage") == 0)
+        {
+            usage = attr.value();
+        }
+        else if ( strcmp(attr.name(),"canID" ) == 0 )
+        {
+            int iVal = attr.as_int();
+            if (iVal > -1 && iVal < 63)
+            {
+                canID = attr.as_int();
+            }
+            else
+            {
+                Logger::GetLogger()->LogError ("==>> CanCoderDefn::ParseXML invalid CAN Id \n", "iVal");
+                hasError = true;
+            }
+        }
+        else
+        {
+            Logger::GetLogger()->LogError ( "CanCoderDefn", "invalid attribute");
+            hasError = true;
+        }
+
+    }   
+    if(!hasError)
+    {
+        cancoder = make_shared<CANCoder>(canID); //need to add usage also can't use new CANCoder... because i am using a shared pointer when defining CANCoder
```

Comment:
this isn't correct it is because you are using a raw cancoder and it isn't wrapped as a dragoncancoder